frontends/p4/staticAssert.h

```diff
@@ -63,13 +65,28 @@ class DoStaticAssert : public Inspector {
                         }
                         ::error(ErrorType::ERR_EXPECTED, "%1%: %2%",
                                 method, message);
+                        return method;
                     }
                 } else {
-                    ::warning("Could not evaluate static_assert to a constant: %1%", arg);
+                    ::error(ErrorType::ERR_UNEXPECTED,
+                            "Could not evaluate static_assert to a constant: %1%", arg);
+                    return method;
                 }
             }
         }
-        return false;
+        if (getContext()->node->is<IR::MethodCallStatement>()) {
```

Comment:
It might be useful to add a testcase for the above case.
Something like:
```
#include <core.p4>
void f()
{
  static_assert(static_assert(true, ""));
}

control c()
{
  apply {
  f();
  }
}
control ct();
package pt(ct c);
pt(c()) main;
```
Currently this produces the following warning I think:
```
static_assert5.p4:4: warning: Could not evaluate static_assert to a constant: static_assert
```
Note I have a `hack` in the code to change the warning/error message format to be more like the GNU message style (I might submit a patch later on to change to that style based on a command line option/environment variable and an option to disable printing out the carrot; this is useful for testing using dejagnu).

Reply:
This test passes with the new changes. I have updated the test case in the PR.
The patch will be much appreciated.